src/main/java/com/buddy/tutorial/service/CurrencyServiceImpl.java

```diff
@@ -0,0 +1,82 @@
+package com.buddy.tutorial.service;
+
+import com.buddy.tutorial.model.CountryCurrencyInfo;
+import com.buddy.tutorial.model.CountryCurrencyResponse;
+import com.buddy.tutorial.model.CurrencyInfo;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+@Slf4j
+public class CurrencyServiceImpl implements CurrencyService {
+
+
+    private final RestTemplate restTemplate = new RestTemplate();
+    @Value("${countriesnow.currency.url}")
+    private String url;
+
+    @Override
+    public List<CurrencyInfo> getCurrencyInfo() {
+        CountryCurrencyResponse response = restTemplate.getForObject(url, CountryCurrencyResponse.class);
+        log.debug("Response from countriesnow {}", response);
+        List<CountryCurrencyInfo> currencyInfoList = null;
+        if (response != null) {
+            currencyInfoList = response.getData();
+        }
+        if (currencyInfoList != null) {
+            return convertDTO(currencyInfoList);
+        }
+        throw new RuntimeException("no response");
+    }
+
+    @Override
+    public CurrencyInfo getCurrencyData(String code) {
+
+        CountryCurrencyResponse response = restTemplate.getForObject(url, CountryCurrencyResponse.class);
+        List<CountryCurrencyInfo> data = null;
+        if (response != null) {
```

Comment:
what do we do if response is null? look at the other endpoint. we handled null cases there